src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs

```diff
@@ -437,5 +442,34 @@ internal void ScriptBackup(BackupOperation backupOperation)
         {
             backupOperation.Execute(TaskExecutionMode.Script);
         }
+
+        internal async Task HandleCreateSasRequest(
+           CreateSasParams optionsParams,
+           RequestContext<CreateSasResponse> requestContext)
+        {
+            try
+            {
+                ConnectionInfo connInfo;
+                bool supported = IsBackupRestoreOperationSupported(optionsParams.OwnerUri, out connInfo);
+                var response = new CreateSasResponse();
+
+                if (supported && connInfo != null)
```

Comment:
Shouldn't you send an error if it's not supported or we can't get a connection?

Reply:
This is how we handle backup operation too

Reply:
Is it better now?